stackslib/src/chainstate/nakamoto/miner.rs

```diff
@@ -498,6 +499,58 @@ impl NakamotoBlockBuilder {
 
         Ok((block, consumed, size))
     }
+
+    /// Make a StackerDB chunk message containing a proposed block.
+    /// Sign it with the miner's private key.
+    /// Automatically determine which StackerDB slot to use.
+    /// Returns Some(chunk) if the given key corresponds to one of the expected miner slots
+    /// Returns None if not
+    /// Returns an error on signing or DB error
+    pub fn make_stackerdb_block_proposal(
```

Comment:
Should this be creating a serialized NakamotoBlockProposal rather than just a serialized block?

Reply:
Got my answer! No, the miners should just propose a NakamotoBlock and the Stacks-Signer should fill in the remaining info for the "NakamotoBlockProposal" struct in the endpoint PR as their view of the chain is the "most valid" one (70% need to agree on their view of the chain state).